frontend/components/SignupForm/SignupForm.tsx

```diff
@@ -1,98 +1,90 @@
 import React from 'react'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
+import { useForm, ErrorMessage } from 'react-hook-form'
 import { trackCreateAccount } from '../../events/users'
-import validateAuth, { IErrors } from '../../lib/validateAuth'
 import { useCreateUserMutation } from '../../generated/graphql'
-import useFormValidation from '../../hooks/useFormValidation'
 import FormError from '../FormError'
 import Button from '../../elements/Button'
 import { brandBlue } from '../../utils'
-
-interface IFormValues {
-  handle: string
-  email: string
-  password: string
-}
-
-const initialState: IFormValues = {
-  handle: '',
-  email: '',
-  password: '',
-}
+import theme from '../../theme'
 
 const SignupForm: React.FC = () => {
-  const { handleChange, values, handleValidate, handleBlur, errors, setValues } = useFormValidation<
-    IFormValues,
-    IErrors
-  >(initialState, validateAuth)
-
-  const [createUser, { loading, error }] = useCreateUserMutation({
-    onCompleted: () => {
-      trackCreateAccount()
-      setValues(initialState)
-    },
+  const router = useRouter()
+  const { handleSubmit, register, errors } = useForm({
+    mode: 'onBlur',
   })
 
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    handleValidate(e)
+  const fieldErrorName = Object.keys(errors)[0] || ''
+
+  const [createUser, { loading, error }] = useCreateUserMutation()
+
+  const onSubmit = (data: any) => {
     if (!loading && Object.keys(errors).length === 0) {
       createUser({
         variables: {
-          handle: values.handle,
-          email: values.email,
-          password: values.password,
+          handle: data.handle,
+          email: data.email,
+          password: data.password,
         },
       })
+      trackCreateAccount()
+      router.push({
+        pathname: '/dashboard/my-feed',
+      })
     }
   }
 
   return (
-    <form onSubmit={handleSubmit}>
+    <form onSubmit={handleSubmit(onSubmit)}>
       <fieldset disabled={loading} aria-busy={loading}>
         <h2>Sign up for an account</h2>
         {error && <FormError error={error} />}
         <label htmlFor="handle">
-          Username
-          <p>{errors?.handle}</p>
+          Display Name
           <input
             type="text"
             name="handle"
-            value={values.handle}
-            placeholder="Your username"
+            placeholder="Your name, or perhaps a fun pseudonym!"
             autoComplete="on"
-            onChange={handleChange}
-            onBlur={handleBlur}
+            ref={register({
+              required: 'Display Name is required',
+              minLength: { value: 6, message: 'Your display name must be at least 6 characters' },
```

Comment:
😱 I wonder if there's a way for me to have a special, shorter display name - just "Nick" 🤔 🤣

Reply:
lol for now I just put 6 chars but we can chat about what it should be 😉

Reply:
@nlitwin made it 3 chars for ya! That feels reasonable for a display name 😌

Reply:
Woooo hoooo!!